test-case/multiple-pause-resume.sh

```diff
@@ -86,9 +86,9 @@ do
     fi
 done
 
-# get the min value of TPLG:'pipeline count' with Case:'pipeline count'
-[[ ${#pipeline_idx_lst[*]} -gt ${OPT_VAL['c']} ]] && max_count=${OPT_VAL['c']} || max_count=${#pipeline_idx_lst[*]}
-[[ $max_count -eq 1 ]] && dlogw "pipeline count is 1, don't need to run this case" && exit 2
+max_count="${OPT_VAL['c']}"
+[ "$max_count" -lt "${#pipeline_idx_lst[*]}" ] && [ "$max_count" != "1" ] ||
+    max_count=${#pipeline_idx_lst[*]}
```

Comment:
This is confusing because it looks like if/then/else but it is not equivalent to if/then/else, compare these two:
```shell
if true; then false; else echo REAL_ELSE;     fi    # does not print anything
   true  &&   false   ||  echo NOT_LIKE_ELSE        # prints NOT_LIKE_ELSE !
```

Avoid using `&&` alone like this, it can also cause problems with `set -e`:
```shell
 set -e ;   false && true;   echo HELLO )    # Prints HELLO
( set -e ; ( false && true) ; echo HELLO )    # Does NOT print HELLO!
```

Explanations:

https://www.shellcheck.net/wiki/SC2015
https://mywiki.wooledge.org/BashGuide/TestsAndConditionals#Control_Operators_.28.26.26_and_.7C.7C.29
http://redsymbol.net/articles/unofficial-bash-strict-mode/#short-circuiting


`||` is great for very simple things like `do_this || die` but for anything more complex it's better to fallback on the "English" and usual if/then/else.

Reply:
let's use if.

Reply:
```
In test-case/multiple-pause-resume.sh line 90:
if [ "$max_count" -gt "$PIPELINE_COUNT" -o "$max_count" == "1" ]; then
                                        ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.
```
https://github.com/koalaman/shellcheck/wiki/SC2166